test/files/run/t9915/C_1.java

```diff
@@ -0,0 +1,18 @@
+
+public class C_1 {
+    public static final String NULLED = "X\000ABC";
```

Comment:
This is a pretty good selection of test cases. Just to ensure complete parity with the Java spec, I wonder if it'll be a good idea to add additional test cases and asserts?

Set 1 : asserts that test encoded single byte, 2 byte and 3 bytes of zeros
public class TestNull {
  public static  String null1 = "A\0B"; <- access from scala code and verify that it should be of length 3
  public static  String null2 = "A\00B"; <- should be of length 3
  public static  String null3 = "A\000B"; <- should be of length 3
  public static   String null4 = "A\0000B"; <- should be of length 4
  public static  String null5 = "A\u0000B"; <- should be of length 3
}

It wonder if we could verify TestNull.null1 == TestNull.null2 == TestNull.null3 == TestNull.null5 == "A\0B" == "A\u0000B" from scala code? 

Please let me know if the additional asserts will be useful (or may be an overkill).